### ISSUE TITLE
General Memory Leak in Network Visualization

### ISSUE THREAD

**Issue by hbaderhold** [almende/vis#533]
Hi,

I'm completely new to coding and javascript, but I find vis.js absolutely awesome and very intuitive to use!

I do have a general problem with the dynamic network visualization. Example: In your Gallery http://visjs.org/examples/network/02_random_nodes.html if you enter 'Number of nodes:' [1000] and hit 'go' a few [2-3] times and then enter 'Number of nodes:' [10] directly after that, the visualization has a significant lag (e.g. wenn dragging nodes). If you refresh the browser, it works fine again. Is this due to a memory leak? If yes, how can I solve this best (in layman's terms)?

(I'm using Safari Version 7.0.6 (9537.78.2); Chrome Version 39.0.2171.95 (64-bit); Firefox 34.0.5. on Mac OS 10.9.4)

Thanks a lot and belated happy new year to the almende team
Hanns

**Comment by AlexDM0** [comment]
Hi,

I noticed this as well a while back. We're in the process of looking into the correct cleaning up of the network on delete. The examples implement this badly. The best way to do this at the moment, is to use a dataset and clear/add the data there WITHOUT re-initializing the network. There is documentation available on the dataset on our website and it is being used in a lot of the examples.

Regards,

Alex

**Comment by AlexDM0** [comment]
Hi Hanns,

I have pushed a fix for this to the develop branch. We'll release this soon.

Regards,

Alex

**Comment by beeva-ivanmarques** [comment]
Is there any news on this topic?

**Comment by mojoaxel** [comment]
@beeva-ivanmarques This is a very old issue. If you still think this is a problem please open a new issue.

**Comment by fatore** [comment]
I'm facing a similar issue: https://github.com/almende/vis/issues/3439

**Comment by wimrijnders** [comment]
@fatore This is not a similar issue; he's talking about lag here while dragging, and the *hypothesis* is that this is caused due to memory leakage. In any case, the issue has been closed for over two years. Please concentrate on your own issue.

**Comment by fatore** [comment]
Indeed, I only commented here to leave a reference to the other issue since searches for this problem are leading here.
